chemprop/args.py

```diff
@@ -546,14 +582,37 @@ def process_args(self) -> None:
                 for key, value in config.items():
                     setattr(self, key, value)
 
+        # Determine the target_columns when training atomic and bond targets
+        self.is_atom_bond_targets = False
+        if self.atom_targets != [] or self.bond_targets != []:
+            self.is_atom_bond_targets = True
+            self.target_columns = self.atom_targets + self.bond_targets
+
+        # Check whether atom/bond constraints have been applied on the correct dataset_type
+        if self.constraints_path:
+            if not self.is_atom_bond_targets:
+                raise ValueError('Constraints on atomic/bond targets can only be used in atomic/bond properties prediction.')
+            if self.dataset_type != 'regression':
+                raise ValueError(f'In atomic/bond properties prediction, atomic/bond constraints are not supported for {self.dataset_type}.')
+
+        # Check whether the number of input columns is one for the atomic/bond mode
+        if self.is_atom_bond_targets:
+            if len(self.smiles_columns) != 1:
+                raise ValueError('In atomic/bond properties prediction, exactly one smiles column must be provided.')
+
+        # Check whether the argument of `--adding_h` is used
+        if self.is_atom_bond_targets:
+            if not self.adding_h:
+                raise ValueError('In atomic/bond properties prediction, argument `--adding_h` should be used.')
```

Comment:
Should this always be the case? Couldn't you have places where people are just training off of heavy atoms?

Reply:
It is a good point. If people want to train on heavy atoms, this error message doesn't make sense. This code would be removed. Besides, I would add some descriptions in README.md to notice people the argument of `--adding_h` should be used when training on atoms including hydrogen atoms.